lib/dynamic_config/dcdo.rb

```diff
@@ -69,9 +69,11 @@ def self.create
     env = Rails.env.to_s if defined?(Rails) && Rails.respond_to?(:env)
 
     cache_expiration = 5
-    if env == 'test'
+    # Use the memory adapter if we're running unit tests, but not if we're running the web application server.
+    if env == 'test' && File.basename($0) != 'puma'
       adapter = MemoryAdapter.new
-    elsif env == 'production'
+    # Production and the managed test system web application servers (test.code.org / studio.code.org) use DynamoDB.
+    elsif env == 'production' || (CDO.test_system? && File.basename($0) == 'puma')
```

Comment:
Note that `CDO.test_system?` does not support `staging-next`. Should we update that method to return true for staging-next?